src/AbstractAlgebra.jl

```diff
@@ -409,7 +409,7 @@ macro show_name(io, O)
         set_name!(o, s)
       end
     end
-    if get(i, :compact, false) && s !== nothing
+    if (get(i, :supercompact, false) || get(i, :compact, false)) && s !== nothing
```

Comment:
If you end up making any other changes, perhaps also do this micro optimization (no need to query the properties of `i` if `s` is `nothing` anyway
```suggestion
    if s !== nothing && (get(i, :supercompact, false) || get(i, :compact, false))
```
But it's not worth it restarting CI just for this.

Reply:
I've added this suggestion as a commit locally with a `[skip ci]` remark. I will push it as soon as current CI run is done.